build.go

```diff
@@ -41,64 +92,80 @@ func Build(
 			logger.Subprocess("Please specify the project path through the $BP_DOTNET_PROJECT_PATH environment variable instead. See README.md or the documentation on paketo.io for more information.")
 		}
 
-		config, err := configParser.Parse(filepath.Join(context.WorkingDir, "*.runtimeconfig.json"))
+		runtimeConfig, err := configParser.Parse(filepath.Join(context.WorkingDir, "*.runtimeconfig.json"))
 		if err != nil {
 			return packit.BuildResult{}, fmt.Errorf("failed to find *.runtimeconfig.json: %w", err)
 		}
 
-		command := filepath.Join(context.WorkingDir, config.AppName)
+		logger.GeneratingSBOM(context.WorkingDir)
+		var sbomContent sbom.SBOM
+		duration, err := clock.Measure(func() error {
+			sbomContent, err = sbomGenerator.Generate(context.WorkingDir)
+			return err
+		})
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		logger.Action("Completed in %s", duration.Round(time.Millisecond))
+		logger.Break()
+
+		logger.FormattingSBOM(context.BuildpackInfo.SBOMFormats...)
+		sbomFormatter, err := sbomContent.InFormats(context.BuildpackInfo.SBOMFormats...)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		command := filepath.Join(context.WorkingDir, runtimeConfig.AppName)
 		var args []string
-		if !config.Executable {
-			_, err := os.Stat(filepath.Join(context.WorkingDir, fmt.Sprintf("%s.dll", config.AppName)))
+		if !runtimeConfig.Executable {
+			_, err := os.Stat(filepath.Join(context.WorkingDir, fmt.Sprintf("%s.dll", runtimeConfig.AppName)))
 			if err != nil && !errors.Is(err, os.ErrNotExist) {
 				return packit.BuildResult{}, err
 			}
 			if errors.Is(err, os.ErrNotExist) {
-				return packit.BuildResult{}, fmt.Errorf("no entrypoint [%s.dll] found: %w ", config.AppName, err)
+				return packit.BuildResult{}, fmt.Errorf("no entrypoint [%s.dll] found: %w ", runtimeConfig.AppName, err)
 			}
 
 			command = "dotnet"
-			args = append(args, fmt.Sprintf("%s.dll", filepath.Join(context.WorkingDir, config.AppName)))
+			args = append(args, fmt.Sprintf("%s.dll", filepath.Join(context.WorkingDir, runtimeConfig.AppName)))
 		}
 
+		// Strip '.' from app name to avoid bug in lifecycle
+		// See https://github.com/buildpacks/lifecycle/issues/895
```

Comment:
Do we have a nice way to track all of these issues (this one and the go-env one) so we can remove logic when the bugs are fixed?

Reply:
Good point. I'll file issues on this repo so we don't lose track of those pieces of tech debt.

Reply:
Filed https://github.com/paketo-buildpacks/dotnet-execute/issues/350 and https://github.com/paketo-buildpacks/dotnet-execute/issues/351